### DISCUSSION TITLE
CallbackURL being ignored?

### DISCUSSION THREAD

**Discussion by apeck14** [nextauthjs/next-auth#6907] [Help]
### Question 💬

When passing in `callbackUrl` to `signIn()` or `signOut()`, it seems to simply be ignored. I have exhausted many of the workarounds others have tried, but it seems nothing is working. I am using the default `redirect` callback in `/api/auth/[...nextAuth.js]`. When I check the cookies set in the browser, it appears to never be updated to the correct value (the `callbackUrl` passed). 

Cookie:
`next-auth.callback-url: "http://localhost:3000/me"`

### How to reproduce ☕️

```js
<SignOutBtn onClick={() => signOut("discord", { callbackUrl: "/" )}>Sign Out</SignOutBtn>
```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by Maclay74** [comment]
The same problem. I use it with credentials provider and it just simply ignored.
No matter what I do, it's always "https://auth" which is obviously very stupid.

**Comment by hampak** [comment]
Hello, I was facing the same issue. I'm using a cloud IDE and cloud IDE's URL paths are not named "http://localhost:3000".

While searching for a solution @balazsorban44 said that **fixing** the redirect URL to `localhost:3000` was intentional (check it out here: https://github.com/nextauthjs/next-auth/pull/7038"

He says that he's going to come up with a solution so that we can modify the redirect URLs accordingly :D
